addons/sale_timesheet/models/sale_order.py

```diff
@@ -19,7 +19,8 @@ class SaleOrder(models.Model):
 
     visible_project = fields.Boolean('Display project', compute='_compute_visible_project', readonly=True)
     project_id = fields.Many2one(
-        'project.project', 'Project', domain="[('billable_type', 'in', ('no', 'task_rate')), ('analytic_account_id', '!=', False), ('company_id', '=', company_id)]",
+        'project.project', 'Project', check_company=True,
+        domain="[('billable_type', 'in', ('no', 'task_rate')), ('analytic_account_id', '!=', False), ('company_id', '=', company_id)]",
```

Comment:
not sure since readonly, to check with @jem-odoo

Reply:
Even if readonly, the domain only has UI purposes if the field is shown AFAIK

Reply:
It is readonly only when the SO is confirmed. 
I think we should handle the cross company task creation, so my domain is probably wrong, and the check_company is not needed. I'm not 100% sure, as I still fear some access rights error later in the flow.

Reply:
i'll leave the check_company out for now, can always be added by bugfix later on and not a terribly difficult behaviour to correct afterwards if needed